// @metamask/controller-utils/dist/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});



var _chunkM6G3ISXGjs = require('./chunk-M6G3ISXG.js');
require('./chunk-DBJSLKNR.js');


























var _chunkCQEHTNYTjs = require('./chunk-CQEHTNYT.js');























var _chunkAEMFFAHNjs = require('./chunk-AEMFFAHN.js');








var _chunkLGGBBU6Gjs = require('./chunk-LGGBBU6G.js');


























































exports.ASSET_TYPES = _chunkAEMFFAHNjs.ASSET_TYPES; exports.ApprovalType = _chunkAEMFFAHNjs.ApprovalType; exports.BNToHex = _chunkCQEHTNYTjs.BNToHex; exports.BUILT_IN_NETWORKS = _chunkAEMFFAHNjs.BUILT_IN_NETWORKS; exports.BuiltInNetworkName = _chunkLGGBBU6Gjs.BuiltInNetworkName; exports.CHAIN_ID_TO_ETHERS_NETWORK_NAME_MAP = _chunkAEMFFAHNjs.CHAIN_ID_TO_ETHERS_NETWORK_NAME_MAP; exports.ChainId = _chunkLGGBBU6Gjs.ChainId; exports.ERC1155 = _chunkAEMFFAHNjs.ERC1155; exports.ERC1155_INTERFACE_ID = _chunkAEMFFAHNjs.ERC1155_INTERFACE_ID; exports.ERC1155_METADATA_URI_INTERFACE_ID = _chunkAEMFFAHNjs.ERC1155_METADATA_URI_INTERFACE_ID; exports.ERC1155_TOKEN_RECEIVER_INTERFACE_ID = _chunkAEMFFAHNjs.ERC1155_TOKEN_RECEIVER_INTERFACE_ID; exports.ERC20 = _chunkAEMFFAHNjs.ERC20; exports.ERC721 = _chunkAEMFFAHNjs.ERC721; exports.ERC721_ENUMERABLE_INTERFACE_ID = _chunkAEMFFAHNjs.ERC721_ENUMERABLE_INTERFACE_ID; exports.ERC721_INTERFACE_ID = _chunkAEMFFAHNjs.ERC721_INTERFACE_ID; exports.ERC721_METADATA_INTERFACE_ID = _chunkAEMFFAHNjs.ERC721_METADATA_INTERFACE_ID; exports.FALL_BACK_VS_CURRENCY = _chunkAEMFFAHNjs.FALL_BACK_VS_CURRENCY; exports.GANACHE_CHAIN_ID = _chunkAEMFFAHNjs.GANACHE_CHAIN_ID; exports.GWEI = _chunkAEMFFAHNjs.GWEI; exports.IPFS_DEFAULT_GATEWAY_URL = _chunkAEMFFAHNjs.IPFS_DEFAULT_GATEWAY_URL; exports.InfuraNetworkType = _chunkLGGBBU6Gjs.InfuraNetworkType; exports.MAX_SAFE_CHAIN_ID = _chunkAEMFFAHNjs.MAX_SAFE_CHAIN_ID; exports.NetworkType = _chunkLGGBBU6Gjs.NetworkType; exports.NetworksTicker = _chunkLGGBBU6Gjs.NetworksTicker; exports.OPENSEA_PROXY_URL = _chunkAEMFFAHNjs.OPENSEA_PROXY_URL; exports.ORIGIN_METAMASK = _chunkAEMFFAHNjs.ORIGIN_METAMASK; exports.RPC = _chunkAEMFFAHNjs.RPC; exports.TESTNET_TICKER_SYMBOLS = _chunkAEMFFAHNjs.TESTNET_TICKER_SYMBOLS; exports.convertHexToDecimal = _chunkCQEHTNYTjs.convertHexToDecimal; exports.detectSIWE = _chunkM6G3ISXGjs.detectSIWE; exports.fetchWithErrorHandling = _chunkCQEHTNYTjs.fetchWithErrorHandling; exports.fractionBN = _chunkCQEHTNYTjs.fractionBN; exports.fromHex = _chunkCQEHTNYTjs.fromHex; exports.getBuyURL = _chunkCQEHTNYTjs.getBuyURL; exports.gweiDecToWEIBN = _chunkCQEHTNYTjs.gweiDecToWEIBN; exports.handleFetch = _chunkCQEHTNYTjs.handleFetch; exports.hexToBN = _chunkCQEHTNYTjs.hexToBN; exports.hexToText = _chunkCQEHTNYTjs.hexToText; exports.isInfuraNetworkType = _chunkLGGBBU6Gjs.isInfuraNetworkType; exports.isNetworkType = _chunkLGGBBU6Gjs.isNetworkType; exports.isNonEmptyArray = _chunkCQEHTNYTjs.isNonEmptyArray; exports.isPlainObject = _chunkCQEHTNYTjs.isPlainObject; exports.isSafeChainId = _chunkCQEHTNYTjs.isSafeChainId; exports.isSmartContractCode = _chunkCQEHTNYTjs.isSmartContractCode; exports.isValidHexAddress = _chunkCQEHTNYTjs.isValidHexAddress; exports.isValidJson = _chunkCQEHTNYTjs.isValidJson; exports.isValidSIWEOrigin = _chunkM6G3ISXGjs.isValidSIWEOrigin; exports.normalizeEnsName = _chunkCQEHTNYTjs.normalizeEnsName; exports.parseDomainParts = _chunkM6G3ISXGjs.parseDomainParts; exports.query = _chunkCQEHTNYTjs.query; exports.safelyExecute = _chunkCQEHTNYTjs.safelyExecute; exports.safelyExecuteWithTimeout = _chunkCQEHTNYTjs.safelyExecuteWithTimeout; exports.successfulFetch = _chunkCQEHTNYTjs.successfulFetch; exports.timeoutFetch = _chunkCQEHTNYTjs.timeoutFetch; exports.toChecksumHexAddress = _chunkCQEHTNYTjs.toChecksumHexAddress; exports.toHex = _chunkCQEHTNYTjs.toHex; exports.weiHexToGweiDec = _chunkCQEHTNYTjs.weiHexToGweiDec;
