// @metamask/snaps-cli/dist/config.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});










var _chunkAKAP22ELjs = require('./chunk-AKAP22EL.js');
require('./chunk-S45DIQPN.js');
require('./chunk-4QUVE3EK.js');
require('./chunk-C46KEPAC.js');
require('./chunk-PDYQXXLP.js');
require('./chunk-DBLADLQ4.js');
require('./chunk-XGWLFH7Y.js');
require('./chunk-B3NNVTA6.js');
require('./chunk-TJ2F3J6X.js');
require('./chunk-S7UP2YYF.js');
require('./chunk-BALQOCUO.js');
require('./chunk-X3D3MHWF.js');











exports.LegacyOptionsStruct = _chunkAKAP22ELjs.LegacyOptionsStruct; exports.SnapsBrowserifyConfigStruct = _chunkAKAP22ELjs.SnapsBrowserifyConfigStruct; exports.SnapsConfigStruct = _chunkAKAP22ELjs.SnapsConfigStruct; exports.SnapsWebpackConfigStruct = _chunkAKAP22ELjs.SnapsWebpackConfigStruct; exports.getConfig = _chunkAKAP22ELjs.getConfig; exports.getConfigByArgv = _chunkAKAP22ELjs.getConfigByArgv; exports.getWebpackConfig = _chunkAKAP22ELjs.getWebpackConfig; exports.loadConfig = _chunkAKAP22ELjs.loadConfig; exports.mergeLegacyOptions = _chunkAKAP22ELjs.mergeLegacyOptions; exports.resolveConfig = _chunkAKAP22ELjs.resolveConfig;
