// @metamask/snaps-cli/dist/webpack/utils.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});











var _chunkJEMJNDZOjs = require('../chunk-JEMJNDZO.js');
require('../chunk-QHCKVO3P.js');
require('../chunk-YQ2E7NEA.js');
require('../chunk-4QUVE3EK.js');
require('../chunk-C46KEPAC.js');
require('../chunk-PDYQXXLP.js');
require('../chunk-DBLADLQ4.js');
require('../chunk-XGWLFH7Y.js');
require('../chunk-B3NNVTA6.js');
require('../chunk-TJ2F3J6X.js');
require('../chunk-S7UP2YYF.js');
require('../chunk-BALQOCUO.js');
require('../chunk-X3D3MHWF.js');












exports.BROWSERSLIST_FILE = _chunkJEMJNDZOjs.BROWSERSLIST_FILE; exports.WEBPACK_FALLBACKS = _chunkJEMJNDZOjs.WEBPACK_FALLBACKS; exports.formatText = _chunkJEMJNDZOjs.formatText; exports.getBrowserslistTargets = _chunkJEMJNDZOjs.getBrowserslistTargets; exports.getDefaultLoader = _chunkJEMJNDZOjs.getDefaultLoader; exports.getDevTool = _chunkJEMJNDZOjs.getDevTool; exports.getEnvironmentVariables = _chunkJEMJNDZOjs.getEnvironmentVariables; exports.getFallbacks = _chunkJEMJNDZOjs.getFallbacks; exports.getImageSVG = _chunkJEMJNDZOjs.getImageSVG; exports.getProgressHandler = _chunkJEMJNDZOjs.getProgressHandler; exports.pluralize = _chunkJEMJNDZOjs.pluralize;
