// @metamask/snaps-controllers/dist/services/browser.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});require('../chunk-A6WJEA3C.js');


var _chunkYVNRYLOCjs = require('../chunk-YVNRYLOC.js');
require('../chunk-C4D6GBMY.js');
require('../chunk-OHMCPTOI.js');
require('../chunk-D74XJG2L.js');


var _chunk3ZC4LB7Tjs = require('../chunk-3ZC4LB7T.js');
require('../chunk-HOY6NKQR.js');


var _chunkZ6SOB3SYjs = require('../chunk-Z6SOB3SY.js');


var _chunkWQYMF7QPjs = require('../chunk-WQYMF7QP.js');
require('../chunk-NSLYI2LS.js');



var _chunk54TYYPANjs = require('../chunk-54TYYPAN.js');
require('../chunk-OIEUL55W.js');


var _chunk7Y6P5FRNjs = require('../chunk-7Y6P5FRN.js');
require('../chunk-NXZVKBSV.js');
require('../chunk-XE5IWVNV.js');
require('../chunk-NOLPMHXS.js');
require('../chunk-EXN2TFDJ.js');








exports.AbstractExecutionService = _chunk54TYYPANjs.AbstractExecutionService; exports.IframeExecutionService = _chunkZ6SOB3SYjs.IframeExecutionService; exports.OffscreenExecutionService = _chunkWQYMF7QPjs.OffscreenExecutionService; exports.ProxyPostMessageStream = _chunk7Y6P5FRNjs.ProxyPostMessageStream; exports.WebViewExecutionService = _chunkYVNRYLOCjs.WebViewExecutionService; exports.WebWorkerExecutionService = _chunk3ZC4LB7Tjs.WebWorkerExecutionService; exports.setupMultiplex = _chunk54TYYPANjs.setupMultiplex;
