// @metamask/snaps-controllers/dist/services/node.mjs
import "../chunk-FX5AADOV.mjs";
import "../chunk-G7U6WKWS.mjs";
import {
  NodeProcessExecutionService
} from "../chunk-25LS4KZK.mjs";
import {
  NodeThreadExecutionService
} from "../chunk-Y5KX5ICF.mjs";
import "../chunk-IABOI7TW.mjs";
import "../chunk-ESQPQNEF.mjs";
import "../chunk-V6NFZ47P.mjs";
import {
  WebWorkerExecutionService
} from "../chunk-SCERVW3B.mjs";
import "../chunk-KCMULJ2M.mjs";
import {
  IframeExecutionService
} from "../chunk-BJOQRBH2.mjs";
import {
  OffscreenExecutionService
} from "../chunk-2KCUVQFS.mjs";
import "../chunk-Q4I5QTVC.mjs";
import {
  AbstractExecutionService,
  setupMultiplex
} from "../chunk-WVGK5B5P.mjs";
import "../chunk-3RNW7OKG.mjs";
import {
  ProxyPostMessageStream
} from "../chunk-ZVOYOZFT.mjs";
import "../chunk-4HVWEABQ.mjs";
import "../chunk-36YC4Z3T.mjs";
import "../chunk-VOZOZKRC.mjs";
import "../chunk-YRZVIDCF.mjs";
export {
  AbstractExecutionService,
  IframeExecutionService,
  NodeProcessExecutionService,
  NodeThreadExecutionService,
  OffscreenExecutionService,
  ProxyPostMessageStream,
  WebWorkerExecutionService,
  setupMultiplex
};
