// @metamask/snaps-utils/dist/snaps.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


















var _chunkTG4BDSYJjs = require('./chunk-TG4BDSYJ.js');
require('./chunk-CMOSYNZR.js');
require('./chunk-ASZWZ7JT.js');
require('./chunk-MLOQHUOY.js');
require('./chunk-473MIETW.js');
require('./chunk-HJYRGKCX.js');
require('./chunk-NUCLSR2G.js');
require('./chunk-PHUTP7NB.js');



















exports.BaseSnapIdStruct = _chunkTG4BDSYJjs.BaseSnapIdStruct; exports.HttpSnapIdStruct = _chunkTG4BDSYJjs.HttpSnapIdStruct; exports.LOCALHOST_HOSTNAMES = _chunkTG4BDSYJjs.LOCALHOST_HOSTNAMES; exports.LocalSnapIdStruct = _chunkTG4BDSYJjs.LocalSnapIdStruct; exports.NpmSnapIdStruct = _chunkTG4BDSYJjs.NpmSnapIdStruct; exports.PROPOSED_NAME_REGEX = _chunkTG4BDSYJjs.PROPOSED_NAME_REGEX; exports.ProgrammaticallyFixableSnapError = _chunkTG4BDSYJjs.ProgrammaticallyFixableSnapError; exports.SnapIdStruct = _chunkTG4BDSYJjs.SnapIdStruct; exports.SnapStatus = _chunkTG4BDSYJjs.SnapStatus; exports.SnapStatusEvents = _chunkTG4BDSYJjs.SnapStatusEvents; exports.assertIsValidSnapId = _chunkTG4BDSYJjs.assertIsValidSnapId; exports.getSnapChecksum = _chunkTG4BDSYJjs.getSnapChecksum; exports.getSnapPrefix = _chunkTG4BDSYJjs.getSnapPrefix; exports.isCaipChainId = _chunkTG4BDSYJjs.isCaipChainId; exports.isSnapPermitted = _chunkTG4BDSYJjs.isSnapPermitted; exports.stripSnapPrefix = _chunkTG4BDSYJjs.stripSnapPrefix; exports.validateSnapShasum = _chunkTG4BDSYJjs.validateSnapShasum; exports.verifyRequestedSnapPermissions = _chunkTG4BDSYJjs.verifyRequestedSnapPermissions;
