// @metamask/phishing-controller/dist/PhishingController.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});












var _chunkGAXG2YM5js = require('./chunk-GAXG2YM5.js');













exports.HOTLIST_REFRESH_INTERVAL = _chunkGAXG2YM5js.HOTLIST_REFRESH_INTERVAL; exports.ListKeys = _chunkGAXG2YM5js.ListKeys; exports.ListNames = _chunkGAXG2YM5js.ListNames; exports.METAMASK_HOTLIST_DIFF_FILE = _chunkGAXG2YM5js.METAMASK_HOTLIST_DIFF_FILE; exports.METAMASK_HOTLIST_DIFF_URL = _chunkGAXG2YM5js.METAMASK_HOTLIST_DIFF_URL; exports.METAMASK_STALELIST_FILE = _chunkGAXG2YM5js.METAMASK_STALELIST_FILE; exports.METAMASK_STALELIST_URL = _chunkGAXG2YM5js.METAMASK_STALELIST_URL; exports.PHISHING_CONFIG_BASE_URL = _chunkGAXG2YM5js.PHISHING_CONFIG_BASE_URL; exports.PhishingController = _chunkGAXG2YM5js.PhishingController; exports.STALELIST_REFRESH_INTERVAL = _chunkGAXG2YM5js.STALELIST_REFRESH_INTERVAL; exports.default = _chunkGAXG2YM5js.PhishingController_default; exports.phishingListKeyNameMap = _chunkGAXG2YM5js.phishingListKeyNameMap;
