// @metamask/snaps-jest/dist/chunk-MCYTJUDG.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _chunkXWGECIXIjs = require('./chunk-XWGECIXI.js');


var _chunkTVD4SWH7js = require('./chunk-TVD4SWH7.js');


var _chunkVXPCUDBIjs = require('./chunk-VXPCUDBI.js');



var _chunkDTDYOBCIjs = require('./chunk-DTDYOBCI.js');


var _chunkZWN4SO2Vjs = require('./chunk-ZWN4SO2V.js');


var _chunkWMLSD6B5js = require('./chunk-WMLSD6B5.js');

// src/internals/simulation/simulation.ts
var _basecontroller = require('@metamask/base-controller');
var _jsonrpcmiddlewarestream = require('@metamask/json-rpc-middleware-stream');
var _keytree = require('@metamask/key-tree');





var _node = require('@metamask/snaps-controllers/node');
var _snapsutils = require('@metamask/snaps-utils');
var _readablestream = require('readable-stream');
async function handleInstallSnap(snapId, {
  executionService,
  executionServiceOptions,
  options: rawOptions = {}
} = {}) {
  const options = _chunkVXPCUDBIjs.getOptions.call(void 0, rawOptions);
  const location = _node.detectSnapLocation.call(void 0, snapId, {
    allowLocal: true
  });
  const snapFiles = await _node.fetchSnap.call(void 0, snapId, location);
  const { store, runSaga } = _chunkWMLSD6B5js.createStore.call(void 0, options);
  const controllerMessenger = new (0, _basecontroller.ControllerMessenger)();
  registerActions(controllerMessenger);
  const hooks = getHooks(options, snapFiles, snapId, controllerMessenger);
  const { subjectMetadataController, permissionController } = _chunkDTDYOBCIjs.getControllers.call(void 0, {
    controllerMessenger,
    hooks,
    runSaga,
    options
  });
  const engine = _chunkXWGECIXIjs.createJsonRpcEngine.call(void 0, {
    store,
    hooks,
    permissionMiddleware: permissionController.createPermissionMiddleware({
      origin: snapId
    })
  });
  const ExecutionService = executionService ?? _node.NodeThreadExecutionService;
  const service = new ExecutionService({
    ...executionServiceOptions,
    messenger: controllerMessenger.getRestricted({
      name: "ExecutionService",
      allowedActions: [],
      allowedEvents: []
    }),
    setupSnapProvider: (_snapId, rpcStream) => {
      const mux = _node.setupMultiplex.call(void 0, rpcStream, "snapStream");
      const stream = mux.createStream("metamask-provider");
      const providerStream = _jsonrpcmiddlewarestream.createEngineStream.call(void 0, { engine });
      _readablestream.pipeline.call(void 0, stream, providerStream, stream, (error) => {
        if (error) {
          _snapsutils.logError.call(void 0, `Provider stream failure.`, error);
        }
      });
    }
  });
  await _chunkDTDYOBCIjs.registerSnap.call(void 0, snapId, snapFiles.manifest.result, {
    permissionController,
    subjectMetadataController
  });
  await service.executeSnap({
    snapId,
    sourceCode: snapFiles.sourceCode.toString("utf8"),
    endowments: await _chunkZWN4SO2Vjs.getEndowments.call(void 0, permissionController, snapId)
  });
  return {
    snapId,
    store,
    executionService: service,
    controllerMessenger,
    runSaga
  };
}
function getHooks(options, snapFiles, snapId, controllerMessenger) {
  return {
    getMnemonic: async () => Promise.resolve(_keytree.mnemonicPhraseToBytes.call(void 0, options.secretRecoveryPhrase)),
    getSnapFile: async (path, encoding) => await _chunkTVD4SWH7js.getSnapFile.call(void 0, snapFiles.auxiliaryFiles, path, encoding),
    getIsLocked: () => false,
    createInterface: async (...args) => controllerMessenger.call(
      "SnapInterfaceController:createInterface",
      snapId,
      ...args
    ),
    updateInterface: async (...args) => controllerMessenger.call(
      "SnapInterfaceController:updateInterface",
      snapId,
      ...args
    ),
    getInterfaceState: (...args) => controllerMessenger.call(
      "SnapInterfaceController:getInterface",
      snapId,
      ...args
    ).state
  };
}
function registerActions(controllerMessenger) {
  controllerMessenger.registerActionHandler(
    "PhishingController:maybeUpdateState",
    async () => Promise.resolve()
  );
  controllerMessenger.registerActionHandler(
    "PhishingController:testOrigin",
    () => ({ result: false, type: "all" })
  );
}





exports.handleInstallSnap = handleInstallSnap; exports.getHooks = getHooks; exports.registerActions = registerActions;
