// @metamask/snaps-jest/dist/chunk-Y4DKEF7Y.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});



var _chunkPZDTZGSIjs = require('./chunk-PZDTZGSI.js');

// src/internals/simulation/interface.ts






var _snapssdk = require('@metamask/snaps-sdk');
var _snapsutils = require('@metamask/snaps-utils');
var _effects = require('redux-saga/effects');
function getInterfaceResponse(runSaga, type, content, interfaceActions) {
  switch (type) {
    case _snapssdk.DialogType.Alert:
      return {
        ...interfaceActions,
        type,
        content,
        ok: resolveWith(runSaga, null)
      };
    case _snapssdk.DialogType.Confirmation:
      return {
        ...interfaceActions,
        type,
        content,
        ok: resolveWith(runSaga, true),
        cancel: resolveWith(runSaga, false)
      };
    case _snapssdk.DialogType.Prompt:
      return {
        ...interfaceActions,
        type,
        content,
        ok: resolveWithInput(runSaga),
        cancel: resolveWith(runSaga, null)
      };
    default:
      throw new Error(`Unknown or unsupported dialog type: "${String(type)}".`);
  }
}
function resolveWith(runSaga, value) {
  function* resolveWithSaga() {
    yield _effects.put.call(void 0, _chunkPZDTZGSIjs.resolveInterface.call(void 0, value));
  }
  return async () => {
    await runSaga(resolveWithSaga).toPromise();
  };
}
function resolveWithInput(runSaga) {
  function* resolveWithSaga(value) {
    yield _effects.put.call(void 0, _chunkPZDTZGSIjs.resolveInterface.call(void 0, value));
  }
  return async (value = "") => {
    await runSaga(resolveWithSaga, value).toPromise();
  };
}
function* getStoredInterface(controllerMessenger, snapId) {
  const currentInterface = yield _effects.select.call(void 0, _chunkPZDTZGSIjs.getCurrentInterface);
  if (currentInterface) {
    const { content: content2 } = controllerMessenger.call(
      "SnapInterfaceController:getInterface",
      snapId,
      currentInterface.id
    );
    return { ...currentInterface, content: content2 };
  }
  const { payload } = yield _effects.take.call(void 0, _chunkPZDTZGSIjs.setInterface.type);
  const { content } = controllerMessenger.call(
    "SnapInterfaceController:getInterface",
    snapId,
    payload.id
  );
  return { ...payload, content };
}
function getElement(content, name) {
  const { type } = content;
  if ((type === _snapssdk.NodeType.Button || type === _snapssdk.NodeType.Input) && content.name === name) {
    return { element: content };
  }
  if (_snapsutils.hasChildren.call(void 0, content)) {
    for (const element of content.children) {
      const result = getElement(element, name);
      const form = type === _snapssdk.NodeType.Form ? content.name : result?.form;
      if (result) {
        return { element: result.element, form };
      }
    }
  }
  return void 0;
}
async function clickElement(controllerMessenger, id, content, snapId, name) {
  const result = getElement(content, name);
  _snapssdk.assert.call(void 0, 
    result !== void 0 && result.element.type === _snapssdk.NodeType.Button,
    "No button found in the interface."
  );
  if (result.form && result.element.buttonType === _snapssdk.ButtonType.Submit) {
    const { state } = controllerMessenger.call(
      "SnapInterfaceController:getInterface",
      snapId,
      id
    );
    await controllerMessenger.call(
      "ExecutionService:handleRpcRequest",
      snapId,
      {
        origin: "",
        handler: _snapsutils.HandlerType.OnUserInput,
        request: {
          jsonrpc: "2.0",
          method: " ",
          params: {
            event: {
              type: _snapssdk.UserInputEventType.FormSubmitEvent,
              name: result.form,
              value: state[result.form]
            },
            id
          }
        }
      }
    );
    return;
  }
  if (result.element.buttonType !== _snapssdk.ButtonType.Submit) {
    await controllerMessenger.call(
      "ExecutionService:handleRpcRequest",
      snapId,
      {
        origin: "",
        handler: _snapsutils.HandlerType.OnUserInput,
        request: {
          jsonrpc: "2.0",
          method: " ",
          params: {
            event: {
              type: _snapssdk.UserInputEventType.ButtonClickEvent,
              name: result.element.name
            },
            id
          }
        }
      }
    );
  }
}
function mergeValue(state, name, value, form) {
  if (form) {
    return {
      ...state,
      [form]: {
        ...state[form],
        [name]: value
      }
    };
  }
  return { ...state, [name]: value };
}
async function typeInField(controllerMessenger, id, content, snapId, name, value) {
  const result = getElement(content, name);
  _snapssdk.assert.call(void 0, 
    result !== void 0 && result.element.type === _snapssdk.NodeType.Input,
    "No input found in the interface."
  );
  const { state } = controllerMessenger.call(
    "SnapInterfaceController:getInterface",
    snapId,
    id
  );
  const newState = mergeValue(state, name, value, result.form);
  controllerMessenger.call(
    "SnapInterfaceController:updateInterfaceState",
    id,
    newState
  );
  await controllerMessenger.call("ExecutionService:handleRpcRequest", snapId, {
    origin: "",
    handler: _snapsutils.HandlerType.OnUserInput,
    request: {
      jsonrpc: "2.0",
      method: " ",
      params: {
        event: {
          type: _snapssdk.UserInputEventType.InputChangeEvent,
          name: result.element.name,
          value
        },
        id
      }
    }
  });
}
function* getInterface(runSaga, snapId, controllerMessenger) {
  const { type, id, content } = yield _effects.call.call(void 0, 
    getStoredInterface,
    controllerMessenger,
    snapId
  );
  const interfaceActions = {
    clickElement: async (name) => {
      await clickElement(controllerMessenger, id, content, snapId, name);
    },
    typeInField: async (name, value) => {
      await typeInField(controllerMessenger, id, content, snapId, name, value);
    }
  };
  return getInterfaceResponse(runSaga, type, content, interfaceActions);
}








exports.getInterfaceResponse = getInterfaceResponse; exports.getElement = getElement; exports.clickElement = clickElement; exports.mergeValue = mergeValue; exports.typeInField = typeInField; exports.getInterface = getInterface;
