// @metamask/snaps-jest/dist/internals/simulation/index.mjs
import "../../chunk-5ZFHFICZ.mjs";
import {
  clickElement,
  getElement,
  getInterface,
  getInterfaceResponse,
  mergeValue,
  typeInField
} from "../../chunk-I5P52IMF.mjs";
import {
  getHooks,
  handleInstallSnap,
  registerActions
} from "../../chunk-VSUNWJMQ.mjs";
import "../../chunk-ZG33OGC3.mjs";
import "../../chunk-KNJNL723.mjs";
import "../../chunk-6HHV6YFQ.mjs";
import "../../chunk-Q2OQXAUM.mjs";
import "../../chunk-VOYBI6T4.mjs";
import "../../chunk-IXKO6X55.mjs";
import "../../chunk-FP4H3ADT.mjs";
import "../../chunk-TBDYQSRT.mjs";
import {
  getOptions
} from "../../chunk-IH7UNS5A.mjs";
import {
  DEFAULT_JSON_RPC_ENDPOINT,
  DEFAULT_LOCALE,
  DEFAULT_SRP
} from "../../chunk-6KXCBUNZ.mjs";
import "../../chunk-LG3VKXGH.mjs";
import "../../chunk-NTOY6Y4Q.mjs";
import "../../chunk-TAIJXTLU.mjs";
import "../../chunk-UDOXICJK.mjs";
import "../../chunk-KSIBNOB2.mjs";
import "../../chunk-WH5C5WIZ.mjs";
import "../../chunk-ALRZENWP.mjs";
import "../../chunk-ORO3T6BU.mjs";
import "../../chunk-5U5WB3SM.mjs";
import "../../chunk-RQKT6BQ2.mjs";
import {
  createStore
} from "../../chunk-3M4GRUMH.mjs";
import {
  closeInterface,
  getCurrentInterface,
  resolveInterface,
  setInterface,
  uiSlice
} from "../../chunk-74TIKA2T.mjs";
import "../../chunk-H2464AXT.mjs";
import {
  addNotification,
  clearNotifications,
  getNotifications,
  notificationsSlice,
  removeNotification
} from "../../chunk-LB4R3BUA.mjs";
import {
  clearState,
  getState,
  setState,
  stateSlice
} from "../../chunk-MPZOXW6I.mjs";
import "../../chunk-57SGDM5B.mjs";
import "../../chunk-JMDSN227.mjs";
export {
  DEFAULT_JSON_RPC_ENDPOINT,
  DEFAULT_LOCALE,
  DEFAULT_SRP,
  addNotification,
  clearNotifications,
  clearState,
  clickElement,
  closeInterface,
  createStore,
  getCurrentInterface,
  getElement,
  getHooks,
  getInterface,
  getInterfaceResponse,
  getNotifications,
  getOptions,
  getState,
  handleInstallSnap,
  mergeValue,
  notificationsSlice,
  registerActions,
  removeNotification,
  resolveInterface,
  setInterface,
  setState,
  stateSlice,
  typeInField,
  uiSlice
};
