// @metamask/snaps-jest/dist/internals/simulation/methods/hooks/index.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});require('../../../../chunk-CKRORVDW.js');


var _chunk265BMFM5js = require('../../../../chunk-265BMFM5.js');



var _chunkMJHR5RTYjs = require('../../../../chunk-MJHR5RTY.js');



var _chunkGMTKFAWOjs = require('../../../../chunk-GMTKFAWO.js');


var _chunkRAMAN5RMjs = require('../../../../chunk-RAMAN5RM.js');




var _chunkSB5EPHE3js = require('../../../../chunk-SB5EPHE3.js');
require('../../../../chunk-VHJZERIC.js');
require('../../../../chunk-WMLSD6B5.js');
require('../../../../chunk-PZDTZGSI.js');
require('../../../../chunk-LACTK6EO.js');
require('../../../../chunk-2YE2P5BZ.js');
require('../../../../chunk-LBC2OGSN.js');
require('../../../../chunk-PHUTP7NB.js');










exports.getClearSnapStateMethodImplementation = _chunkSB5EPHE3js.getClearSnapStateMethodImplementation; exports.getCreateInterfaceImplementation = _chunkMJHR5RTYjs.getCreateInterfaceImplementation; exports.getGetInterfaceImplementation = _chunkMJHR5RTYjs.getGetInterfaceImplementation; exports.getGetLocaleMethodImplementation = _chunk265BMFM5js.getGetLocaleMethodImplementation; exports.getGetSnapStateMethodImplementation = _chunkSB5EPHE3js.getGetSnapStateMethodImplementation; exports.getShowDialogImplementation = _chunkRAMAN5RMjs.getShowDialogImplementation; exports.getShowInAppNotificationImplementation = _chunkGMTKFAWOjs.getShowInAppNotificationImplementation; exports.getShowNativeNotificationImplementation = _chunkGMTKFAWOjs.getShowNativeNotificationImplementation; exports.getUpdateSnapStateMethodImplementation = _chunkSB5EPHE3js.getUpdateSnapStateMethodImplementation;
